### PR TITLE
Fix ProxyDrawables not respecting whether the originals would be drawn

### LINKED ISSUE #1283: ProxyDrawable doesn't respect the original drawable's parent's IsPresent state
If a drawable's parent becomes `IsPresent = false`, its proxied version will still be drawn to the screen.

Resolves ppy/osu#1591.

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes #1283, fixes ppy/osu#1591.

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -1346,6 +1346,12 @@ public ProxyDrawable CreateProxy()
             return proxy = new ProxyDrawable(this);
         }
 
+        /// <summary>
+        /// Used exclusively by <see cref="CompositeDrawable.addFromComposite"/> to determine whether this <see cref="Drawable"/> would be drawn,
+        /// for the case where it is drawn through a proxy.
+        /// </summary>
+        internal virtual bool ShouldDrawThroughProxy => IsPartOfComposite && !IsMaskedAway && IsAlive && IsPresent && (Parent?.ShouldDrawThroughProxy ?? true);
```
Comment (Tom94) [inline osu.Framework/Graphics/Drawable.cs]: Wouldn't this recursive property be slow in cases where proxies should be drawn?

Comment (smoogipoo) [reply osu.Framework/Graphics/Drawable.cs]: Well you're basically short-circuiting the entire scene graph from an arbitrary point inside it, and expecting drawables to still respect their parent's states when drawn at this arbitrary position.

I'm open to suggestions on how we can make this faster, or how else to go about this...

**Comment by smoogipoo** [comment]
Don't merge this just yet in any case. It won't work as I'd hoped with chained proxies since they're marked as always not present. Will fix tomorrow.

**Comment by smoogipoo** [comment]
Closing as I'm working on a better solution.

**Pull request closed without merging**